packages/pds/src/api/com/atproto/repo/putRecord.ts

```diff
@@ -122,3 +134,26 @@ export default function (server: Server, ctx: AppContext) {
     },
   })
 }
+
+// WARNING: mutates object
+const updateProfileLegacyBlobRef = async (
+  actorStore: ActorStoreTransactor,
+  record: ProfileRecord,
+) => {
+  if (record.avatar && !record.avatar.original['$type']) {
+    const blob = await actorStore.repo.blob.getBlobMetadata(record.avatar.ref)
```

Comment:
Just double-checking that we like the behavior of throwing here if the blob is not found.

Reply:
Yup think so 👍 

I believe that's the behavior we'd do you tried to update a record and were missing a blob anyways 👍